src/backend/distributed/planner/insert_select_planner.c

```diff
@@ -1355,3 +1373,35 @@ InsertSelectResultIdPrefix(uint64 planId)
 
 	return resultIdPrefix->data;
 }
+
+
+/*
+ * GetInsertSelectMethod returns the preferred INSERT INTO ... SELECT method
+ * based on its select query.
+ */
+static InsertSelectMethod
+GetInsertSelectMethod(Query *selectQuery, Oid targetRelationId, ParamListInfo boundParams)
+{
+	Query *selectQueryCopy = copyObject(selectQuery);
+
+	/*
+	 * Query will be replanned in insert_select_executor, turn off
+	 * logging here to avoid repeated log messages.
+	 */
+	int savedClientMinMessages = client_min_messages;
+	set_config_option("client_min_messages", "ERROR",
```

Comment:
Doing `SET LOCAL` so it is reverted on error. Alternative would have been using a PG_TRY() block.